packages/fxa-js-client/tests/lib/recoveryCodes.js

```diff
@@ -2,144 +2,139 @@
-        };
+const sinon = require('sinon');
+const otplib = require('otplib');
+describe('recovery codes', function() {
+  var account;
+  var accountHelper;
+  var respond;
+  var client;
+  var RequestMocks;
+  var env;
+  var xhr;
+  var xhrOpen;
+  var xhrSend;
+  var recoveryCodes;
+  var metricsContext;
 
-        return accountHelper
-          .newVerifiedAccount()
-          .then(function(newAccount) {
-            account = newAccount;
-            return respond(
-              client.createTotpToken(account.signIn.sessionToken),
-              RequestMocks.createTotpToken
-            );
-          })
-          .then(function(res) {
-            assert.ok(res.qrCodeUrl, 'should return QR code data encoded url');
-            assert.ok(
-              res.secret,
-              'should return secret that is encoded in url'
-            );
+  beforeEach(function() {
+    env = new Environment();
+    accountHelper = env.accountHelper;
+    respond = env.respond;
+    client = env.client;
+    RequestMocks = env.RequestMocks;
+    metricsContext = {
+      flowBeginTime: Date.now(),
+      flowId:
+        '0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef',
+    };
 
-            var authenticator = new otplib.authenticator.Authenticator();
-            authenticator.options = otplib.authenticator.options;
+    return accountHelper
+      .newVerifiedAccount()
+      .then(function(newAccount) {
+        account = newAccount;
+        return respond(
+          client.createTotpToken(account.signIn.sessionToken),
+          RequestMocks.createTotpToken
+        );
+      })
+      .then(function(res) {
+        assert.ok(res.qrCodeUrl, 'should return QR code data encoded url');
+        assert.ok(
+          res.secret,
+          'should return secret that is encoded in url'
+        );
 
-            var code = authenticator.generate(res.secret);
-            return respond(
-              client.verifyTotpCode(account.signIn.sessionToken, code),
-              RequestMocks.verifyTotpCodeTrueEnableToken
-            );
-          })
-          .then(function(res) {
-            assert.equal(
-              res.recoveryCodes.length,
-              8,
-              'should return recovery codes'
-            );
-            recoveryCodes = res.recoveryCodes;
+        var authenticator = new otplib.authenticator.Authenticator();
+        authenticator.options = otplib.authenticator.options;
 
-            xhr = env.xhr;
-            xhrOpen = sinon.spy(xhr.prototype, 'open');
-            xhrSend = sinon.spy(xhr.prototype, 'send');
-          });
-      });
+        var code = authenticator.generate(res.secret);
+        return respond(
+          client.verifyTotpCode(account.signIn.sessionToken, code),
+          RequestMocks.verifyTotpCodeTrueEnableToken
+        );
+      })
+      .then(function(res) {
+        assert.equal(
+          res.recoveryCodes.length,
+          3,
```

Comment:
Do you expect the change from 8 to 3 here?

Reply:
Yeah because 3 is the default in fxa-auth-server dev mode